rust/cloud-storage/properties/src/domain/test.rs

```diff
@@ -2,7 +2,7 @@
 
 use super::service_impl::PropertiesServiceImpl;
 use crate::domain::{
-    ports::{MockPermissionService, MockPropertiesRepo},
+    ports::{MockNotificationService, MockPermissionService, MockPropertiesRepo},
     service::PropertiesService,
 };
 use anyhow::anyhow;
```

Comment:
nit: Your tests have alot of duplicated setup and checking logic. Every time your api changes you will have to change every test.

Write a single `check_notifications(test_case: NotificationTestCase)` fn and just pass in common test cases. WIll make life down the line alot easier.